singularity-eos/eos/eos_SpinerEOSDependsRhoT.hpp

```diff
@@ -0,0 +1,1282 @@
+  } else if (whereAmI == TableStatus::OffTop) { // ideal gas
+    const Real Cv = dEdTMax_.interpToReal(lRho);
+    const Real gm1 = gm1Max_.interpToReal(lRho);
+    const Real e0 = sielTMax_.interpToReal(lRho);
+    const Real e = e0 + Cv * (T - TMax_);
+    P = gm1 * rho * e;
+  } else { // if ( whereAmI == TableStatus::OnTable) {
+    P = P_.interpToReal(lRho, lT);
+  }
+  return P;
+}
+
+PORTABLE_INLINE_FUNCTION
+Real SpinerEOSDependsRhoT::CvFromlRholT_(const Real lRho, const Real lT,
+                                         const TableStatus &whereAmI) const {
+  Real Cv;
+  if (whereAmI == TableStatus::OffBottom) { // cold curve
+    // on cold curve. Currently, we assume constant extrapolation.
+    // TODO(JMM): Do something more sophisticated
+    Cv = dEdTCold_.interpToReal(lRho);
+  } else if (whereAmI == TableStatus::OffTop) { // ideal gas
+    Cv = dEdTMax_.interpToReal(lRho);           // Cv assumed constant in T
+  } else {                                      // on table
+    Cv = dEdT_.interpToReal(lRho, lT);
+  }
+  return Cv > robust::EPS() ? Cv : robust::EPS();
+}
+
+PORTABLE_INLINE_FUNCTION
+Real SpinerEOSDependsRhoT::bModFromRholRhoTlT_(const Real rho, const Real lRho,
+                                               const Real T, const Real lT,
+                                               const TableStatus &whereAmI) const {
+  Real bMod;
+  if (whereAmI == TableStatus::OffBottom) {
+    bMod = bModCold_.interpToReal(lRho);
+  } else if (whereAmI == TableStatus::OffTop) {
+    const Real Cv = dEdTMax_.interpToReal(lRho);
+    const Real gm1 = gm1Max_.interpToReal(lRho);
+    const Real e0 = sielTMax_.interpToReal(lRho);
+    const Real e = e0 + Cv * (T - TMax_);
+    bMod = (gm1 + 1) * gm1 * rho * e;
+  } else { // on table
+    bMod = bMod_.interpToReal(lRho, lT);
+  }
+  return bMod > robust::EPS() ? bMod : robust::EPS();
+}
+
+PORTABLE_INLINE_FUNCTION
+TableStatus SpinerEOSDependsRhoT::getLocDependsRhoSie_(const Real lRho,
+                                                       const Real sie) const {
+  TableStatus whereAmI;
+  const Real sielTMax = sielTMax_.interpToReal(lRho);
+  const Real sieCold = sieCold_.interpToReal(lRho);
+  // sie can be negative, so must make sign right
+  if (sie >= sielTMax - SOFT_THRESH * std::abs(sielTMax)) {
+    whereAmI = TableStatus::OffTop;
+  } else if (sie <= sieCold + SOFT_THRESH * std::abs(sieCold)) {
+    whereAmI = TableStatus::OffBottom;
+  } else {
+    whereAmI = TableStatus::OnTable;
+  }
+  return whereAmI;
+}
+
+PORTABLE_INLINE_FUNCTION TableStatus
+SpinerEOSDependsRhoT::getLocDependsRhoT_(const Real lRho, const Real lT) const {
+  TableStatus whereAmI;
+  if (lT <= (1 + SOFT_THRESH) * lTMin_)
+    whereAmI = TableStatus::OffBottom;
+  else if (lT >= (1 - SOFT_THRESH) * lTMax_)
+    whereAmI = TableStatus::OffTop;
+  else
+    whereAmI = TableStatus::OnTable;
+  return whereAmI;
+}
+
+} // namespace singularity
+
+#endif // _EOS_SPINEREOSDEPENDSRHOT_HPP_
```

Comment:
Add a newline at the end of the file. That's what that red circle means. It's just convention, but it's standard for a file to end with an empty line.

Reply:
Many editors are configurable to do this automatically as well.